lib/matplotlib/colors.py

```diff
@@ -960,6 +960,229 @@ def scaled(self):
         return (self.vmin is not None and self.vmax is not None)
 
 
+class FuncNorm(Normalize):
+    """
+    Creates a normalizer using a custom function
+
+    The normalizer will be a function mapping the data values into colormap
+    values in the [0,1] range.
+    """
+
+    def __init__(self, f, finv=None, **normalize_kw):
```

Comment:
We should probably just follow the example of the other classes; `LogNorm`, `BoundaryNorm` and `NoNorm` only accept `clip` and the rest accept all three explicitly, so being explicit seems to be the best choice.